sdk/document.go

```diff
@@ -16,13 +16,19 @@ package sdk
 
 import "fmt"
 
+// MaxSubCollectionDepth - maximum number of parents a collection can support.
+// Depth is a count of the number of parents for a collection.
+// e.g. a collection with no parent has a depth of 0
+// a collection with a parent has a depth of 1
+const MaxSubCollectionDepth int = 1
+
 type Collection struct {
 	Name   string
 	Parent *Key
 }
 
 type Key struct {
-	Collection Collection
+	Collection *Collection
```

Comment:
Not sure why changing this to be pointer?  Is this to be consistent with Parent being pointer with Collection. From memory perspective its probably more compact as is, but this probably trivial in broader context.

Reply:
It was for consistency with how the Parent is a pointer in the collection, the other is because the method that converts a collection from the pb struct was also returning a pointer. Probably not a big deal either way, like you said.